packages/dialect-react-ui/components/Notifications/screens/NewSettings/Wallet.tsx

```diff
@@ -0,0 +1,154 @@
+import {
+  AddressType,
+  Backend,
+  ThreadMemberScope,
+  useAddresses,
+  useDialectConnectionInfo,
+  useDialectDapp,
+  useThread,
+  useThreads,
+} from '@dialectlabs/react-sdk';
+import { useWallet } from '@solana/wallet-adapter-react';
+import type { PublicKey } from '@solana/web3.js';
+import clsx from 'clsx';
+import { Button, Loader } from '../../../common';
+import { P } from '../../../common/preflighted';
+import { useTheme } from '../../../common/providers/DialectThemeProvider';
+import IconButton from '../../../IconButton';
+
+export const display = (publicKey: PublicKey | string): string => {
+  const s = publicKey.toString();
+  return `${s.slice(0, 4)}...${s.slice(s.length - 4)}`;
+};
+
+const addressType = AddressType.Wallet;
+
+const Wallet = () => {
+  const wallet = useWallet();
+  const { dappAddress } = useDialectDapp();
+  const { textStyles, outlinedInput } = useTheme();
+  const { icons } = useTheme();
+  const { create, isCreatingThread, errorCreatingThread } = useThreads();
+
+  const {
+    addresses: { WALLET: walletObj },
+    addresses,
+    isCreatingAddress,
+    isDeletingAddress,
+  } = useAddresses();
+
+  const {
+    connected: {
+      solana: { shouldConnect: isSolanaShouldConnect },
+      dialectCloud: { shouldConnect: isDialectCloudShouldConnect },
+    },
+  } = useDialectConnectionInfo();
+
+  const isBackendSelectable =
+    isSolanaShouldConnect && isDialectCloudShouldConnect;
+
+  const backend =
+    isSolanaShouldConnect && !isBackendSelectable
+      ? Backend.Solana
+      : Backend.DialectCloud;
+
+  const { thread, delete: deleteDialect } = useThread({
+    findParams: { otherMembers: dappAddress ? [dappAddress] : [] },
+  });
+
+  const { delete: deleteAddress } = useAddresses();
+
+  const deleteThread = async () => {
+    await deleteDialect();
+    await deleteAddress({ addressType });
```

Comment:
hm, I'm not sure that we should delete address, we probably want delete dappAddress, otherwise it might cancel all other subscriptions. @tsmbl please correct me

Reply:
Also was a bit confused. this part was in old settings, but in general agree

Reply:
If you delete address, all subscriptions are deleted across all dapp-sites
This action is pretty destructive but still a valid operation, so I expect at least to have a pop-up with warning or smth like this to communicate consequences to user

@zhelezkov @AlekseiPetrovDev wdyt?

Reply:
thanks for clarifying, highlighting this problem